wpilib/wpilib/nidecbrushless.py

```diff
@@ -0,0 +1,140 @@
+# validated: 2017-12-22 EN f9bece2ffbf7 edu/wpi/first/wpilibj/NidecBrushless.java
+# ----------------------------------------------------------------------------
+#  Copyright (c) 2017 FIRST. All Rights Reserved.                             
+#  Open Source Software - may be modified and shared by FRC teams. The code   
+#  must be accompanied by the FIRST BSD license file in the root directory of 
+#  the project.                                                               
+# ----------------------------------------------------------------------------
+import hal
+from .digitaloutput import DigitalOutput
+from .sendablebase import SendableBase
+from .motorsafety import MotorSafety
+from .pwm import PWM
+from .interfaces.speedcontroller import SpeedController
+
+
+__all__ = ['NidecBrushless']
+
+
+class NidecBrushless(SendableBase, MotorSafety, SpeedController):
+    """Nidec Brushless Motor"""
+
+    def __init__(self, pwmChannel, dioChannel):
+        """
+        :param pwmChannel: The PWM channel that the Nidec Brushless controller is attached to.
+                0-9 are on-board, 10-19 are on the MXP port
+        :param dioChannel: The DIO channel that the Nidec Brushless controller is attached to.
+                0-9 are on-board, 10-25 are on the MXP port
+        """
+        super().__init__()
+        self.dio = DigitalOutput(dioChannel)
+        self.addChild(self.dio)
+        self.dio.setPWMRate(15625)
+        self.dio.enablePWM(0.5)
+
+        self.pwm = PWM(pwmChannel)
+        self.addChild(self.pwm)
+
+        MotorSafety.__init__(self)
+        self.speed = 0.0
+        self.isInverted = False
+        self.disabled = False
+
+        hal.report(hal.UsageReporting.kResourceType_NidecBrushless, pwmChannel)
+        self.setName("Nidec Brushless", pwmChannel)
+
+    def free(self):
+        """Free the resources used by this object."""
+        super().free()
+        self.dio.free()
+        self.pwm.free()
+
+    def set(self, speed):
+        """ 
+        Set the PWM value.
+    
+        The PWM value is set using a range of -1.0 to 1.0, appropriately scaling the value for the FPGA.
+
+       :param speed: The speed value between -1.0 and 1.0 to set.
```

Comment:
Indentation doesn't match.

Reply:
@ariovistus are you converting docstrings by hand? That sounds painful, checkout `devtools/convert_javadoc.html`